zerver/lib/actions.py

```diff
@@ -1963,7 +1963,7 @@ def do_send_messages(
                 flags=user_flags.get(user_id, []),
                 online_push_enabled=(user_id in send_request.online_push_user_ids),
                 stream_push_enabled=(user_id in send_request.stream_push_user_ids),
-                stream_email_notify=(user_id in send_request.stream_email_user_ids),
+                stream_email_enabled=(user_id in send_request.stream_email_user_ids),
                 wildcard_mention_notify=(user_id in send_request.wildcard_mention_user_ids),
```

Comment:
These fields are based on pure configuration, but that configuration includes whether the user has muted the stream or topic.  I worry names like `stream_email_enabled` suggest that it a pure setting, and conceals that structure somewhat.

```
        def should_send(setting: str, row: Dict[str, Any]) -> bool:                        
            # This implements the structure that the UserProfile stream notification settings                                                                                        
            # are defaults, which can be overridden by the stream-level settings (if those 
            # values are not null).                                                        
            if row["is_muted"]:                                                            
                return False                                                               
            if row["user_profile_id"] in user_ids_muting_topic:                            
                return False                                                               
            if row[setting] is not None:                                                   
                return row[setting]                                                        
            return row["user_profile_" + setting]                                          
                                                                                           
        stream_push_user_ids = {                                                           
            row["user_profile_id"]                                                         
            for row in subscription_rows                                                   
            # Note: muting a stream overrides stream_push_notify                           
            if should_send("push_notifications", row)                                      
        }                                                                                  
                                                                                           
        stream_email_user_ids = {                                                          
            row["user_profile_id"]                                                         
            for row in subscription_rows                                                   
            # Note: muting a stream overrides stream_email_notify                          
            if should_send("email_notifications", row)                                     
        }                                                                                  
```

Reply:
I think `is_stream_topic_email_notifiable` would be nice, but maybe its too long?